common/streams/src/stream_limit_by.rs

```diff
@@ -50,20 +52,19 @@ impl LimitByStream {
     }
 
     pub fn limit_by(&mut self, block: &DataBlock) -> Result<Option<DataBlock>> {
-        // TODO: use BitVec here.
```

Comment:
I would suggest to use `arrow2::bitmap::MutableBitmap` here and then use `BooleanArray::from_data(filter_vec.into(), None)`. This should be at least 10x faster, as we avoid running over all bits twice.

```rust
let mut filter_vec = arrow2::bitmap::MutableBitmap::from_zero_len(block.num_rows())
...
filter_vec.set(row as usize, *count <= self.limit)
...

let array = arrow2::array::BooleanArray::from_data(filter_vec.into(), None);
// convert to DFBooleanArray
```

`bitvec` and `arrow2::bitmap` (`MutableBitmap` specifically) overlap in functionality, but creating a bitmap directly from arrow avoids a full scan over all bits and an extra memory allocation. :)

Reply:
Ok, should I open a new PR or modify it in the current one?

Reply:
Modify is the simple way?

Reply:
Ok, I'll fix it.